hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -547,4 +541,42 @@ public void setConf(Configuration conf) {
   public Configuration getConf() {
     return conf;
   }
+
+  @VisibleForTesting
+  HiveClientPool clientPool() {
+    synchronized (CLIENT_POOL_CACHE) {
+      String metastoreUri = conf.get(HiveConf.ConfVars.METASTOREURIS.varname, "");
+      Pair<HiveClientPool, Long> cacheEntry = CLIENT_POOL_CACHE.getIfPresent(metastoreUri);
+      HiveClientPool clientPool = cacheEntry == null ? new HiveClientPool(clientPoolSize, conf) : cacheEntry.first();
```

Comment:
Can we restrict the scope of the synchronised block to contain only the cache population part? that way we could decrease the contention on the more frequent cache lookup operation. If `cacheEntry` is null, we enter the block, and re-check if it's been populated by another thread in the meantime

Reply:
@marton-bod Thanks for the review. I delegated the cache update/cleanup to the cache itself, so now it became threadsafe.